### ISSUE TITLE
Add new IgnoreCaseStep API

### ISSUE THREAD

**Issue by knutwannheden** [jOOQ/jOOQ#8865]
It would be nice with a new `IgnoreCaseStep` (or `LikeIgnoreCaseStep`) API, which would for example be extended by `LikeEscapeStep` and would then allow the user to write:

```java
field.like("foo!%%").escape('!').ignoreCase()
field.like("foo%").ignoreCase()
```

Methods like `Field#likeAny()` would also immediately benefit from this and would thus allow the user to do case-insensitive matching without requiring extra methods.

The new `IgnoreCaseStep` could also be used as the return type for methods like `Field#startsWith()`, which currently has `Condition` as the return type.

In the future this new API should allow deprecating / removing all `Field#xxxIgnoreCase()` methods.

**Comment by knutwannheden** [comment]
Should the `Field#compare(Comparator)` and `Row<N>#compare(Comparator)` methods then also return `IgnoreCaseStep` rather than `Condition`, even though `IGNORE CASE` obviously doesn't make sense for all comparators?

**Comment by lukaseder** [comment]
> Should the Field#compare(Comparator) and Row<N>#compare(Comparator) methods then also return IgnoreCaseStep rather than Condition, even though IGNORE CASE obviously doesn't make sense for all comparators?

I don't think this step should be implemented for `RowN` comparison predicates. In fact, I have some additional doubts about returning this step from `Field<T>.eq(Field<T>)`, as we have no way anymore to connect this step to `<T extends String>` bounds anymore, so the step is being offered way to generically, e.g. for `Field<Integer>.eq(Field<Integer>)`.

Perhaps, we should discuss this once more?

> And would CompareCondition also have to implement IGNORE CASE for the other operators like SIMILAR TO or LESS?

Implementation-wise, this would be trivial. The same rules always apply: Apply `lower()` to both sides of the predicate, and possibly, in the future, offer an SPI that allows for overriding the meaning of `IGNORE CASE`.

The meaning of `lt()` with `IGNORE CASE` is well defined and meaningful. Case-insensitive comparison of strings is useful also outside of `ORDER BY` clauses, and it should behave the same way.
